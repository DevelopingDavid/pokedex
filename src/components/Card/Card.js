import React from 'react';

const Card = ({ pokemonInfo }) => {
  const types = pokemonInfo.types.map(t => {
    const type = t.type.name;
    return type;
  });
  return (
    <article>
      <h1>{pokemonInfo.name}</h1>
      <img src={pokemonInfo.sprite} alt={pokemonInfo.name} />
      <h1>weight: {pokemonInfo.weight}</h1>
      <h1>height: {pokemonInfo.height}</h1>
      {
        types.map(type => {
          return <h1>{type}</h1>
        })
      }
    </article>
  )
}

export default Card;