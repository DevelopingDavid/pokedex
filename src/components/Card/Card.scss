.card {
  font-family: "Flexo-Medium",arial,sans-serif;
  width: 300px;
  height: 300px;
  margin: 10px;
  background: #F2F2F2;
  border-radius: 5px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  img {
    width: 250px;
    height: auto;
  }
  a {
    text-decoration: none;
    color: black;
  }
}

.types-container {
  display: flex;
  justify-content: center;
}