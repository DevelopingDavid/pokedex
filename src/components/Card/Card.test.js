import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

let mockPokemon = {
  name: 'charmander',
  types: [{ slot: 2, type: { name: 'fire' } }]
}

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card key={mockPokemon.name} pokemonInfo={mockPokemon} />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});