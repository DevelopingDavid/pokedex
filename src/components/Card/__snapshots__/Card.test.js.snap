// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapShot 1`] = `
<article
  className="card"
>
  <Link
    to="/pokemon/undefined"
  >
    <img
      alt="charmander"
    />
    <h1>
      charmander
    </h1>
    <div
      className="types-container"
    >
      <Type
        key="fire"
        type="fire"
      />
    </div>
  </Link>
</article>
`;
